### DISCUSSION TITLE
How to use AsyncCheckpointer with CheckpointManager

### DISCUSSION THREAD

**Discussion by davidshen84** [google/orbax#971] [Q&A] [answered]
Hi, 

Based on this code snippet from [this document](https://orbax.readthedocs.io/en/latest/async_checkpointing.html), the manager will automatically figure out how to use the *async checkpointer*. Is that correct?

```python
path = epath.Path('/tmp/async_checkpoint_manager')
ckpt_mngr = ocp.CheckpointManager(path)

def train_step(step, state):
  # update state values accordingly
  return step + 1, state

step = 0
num_steps = 5
while step < num_steps:
  ckpt_mngr.save(step, args=ocp.args.StandardSave(train_state))
  step, train_state = train_step(step, train_state)

ckpt_mngr.wait_until_finished()
```

What about multi-item checkpointing? Do we need to define arguments like the `item_names` and `item_handlers`?

**Comment by cpgaffney1** [comment, marked as answer]
Apologies for not spotting this question, undoubtedly you have resolved the question by now. For posterity though, yes, async saving behavior is used by default. 

A more recent change has made it so that you do not need to define `item_names` and `item_handlers` in advance. For multi-item checkpointing, just specify `args=ocp.args.Composite(state=..., dataset=..., ...)`.

**Comment by davidshen84** [comment]
The document website has changed so much, the link in my original question is gone.

I think [this link](https://orbax.readthedocs.io/en/latest/guides/checkpoint/api_refactor.html#id2) is a suitable replacement and also provides the answer.
